apps/preview/next/pages/showcases/SelectDropdown/SelectDropdownRequired.tsx

```diff
@@ -0,0 +1,118 @@
+    setSelectedOption(option);
+    setListVisible(false);
+    selectDropdownRef.current?.focus();
+  };
+
+  const handleTriggerKeyDown = (e: KeyboardEvent<HTMLDivElement>) => {
+    if (e.key === ' ') setListVisible((isOpen) => !isOpen);
+  };
+
+  const handleOptionItemKeyDown = (e: KeyboardEvent<HTMLLIElement>, option: OptionType) => {
+    if (e.key === ' ' || e.key === 'Enter') selectOption(option);
+  };
+
+  return (
+    <>
+      <SfDropdown
+        open={listVisible}
+        onClose={() => setListVisible(false)}
+        className="relative !w-full"
+        dropdownClassName="w-full"
+        trigger={
+          <label className="font-medium typography-label-sm">
+            Speed
+            <div
+              ref={selectDropdownRef}
+              role="combobox"
+              aria-required="true"
+              aria-controls={listboxId}
+              aria-expanded={listVisible}
+              aria-label="Select one option"
+              aria-activedescendant={selectedOption ? `${listboxId}-${selectedOption.value}` : undefined}
+              className="mt-0.5 flex items-center gap-8 relative font-normal typography-text-base ring-1 ring-neutral-300 ring-inset rounded-md py-2 px-4 hover:ring-primary-700 active:ring-primary-700 active:ring-2 focus-within:ring-primary-700 focus-within:ring-2 cursor-pointer"
+              tabIndex={0}
+              onKeyDown={handleTriggerKeyDown}
+              onClick={() => setListVisible((isOpen) => !isOpen)}
+            >
+              {selectedOption ? selectedOption.label : <span className="text-neutral-500">Choose from the list</span>}
+              <SfIconExpandMore
+                className={classNames('ml-auto text-neutral-500 transition-transform ease-in-out duration-300', {
+                  'rotate-180': listVisible,
+                })}
+              />
+            </div>
+          </label>
+        }
+      >
+        <ul
+          id={listboxId}
+          ref={listboxRef}
+          role="listbox"
+          aria-label="Select one option"
+          className="absolute w-full py-2 rounded-md shadow-md border border-neutral-100 bg-white z-10"
+        >
+          {options.map((option) => (
+            <SfListItem
+              id={`${listboxId}-${option.value}`}
+              key={option.value}
+              role="option"
+              tabIndex={0}
+              aria-selected={option.value === selectedOption?.value}
+              className="block"
+              onClick={() => selectOption(option)}
+              onKeyDown={(e) => handleOptionItemKeyDown(e, option)}
+              slotSuffix={option.value === selectedOption?.value && <SfIconCheck className="text-primary-700" />}
+            >
+              {option.label}
+            </SfListItem>
+          ))}
+        </ul>
+      </SfDropdown>
+      <p className="mt-0.5 text-neutral-500 typography-text-xs">Help text</p>
+      <p className="mt-2 text-neutral-500 typography-text-sm">*Required</p>
```

Comment:
Is this example going to be deleted after todays discussion?

Reply:
nope. it will be handled in a separate task